description: fix a build error on 32-bit platforms

As described in https://bugs.debian.org/1011096 , recent versions of
chromium fail to build on i386 and armhf. This patch, pulled from
upstream, should fix it.



commit e334ce2c23f02f0cc0bfde9e335fd50ec6143b62
Author: Chisoon Jeong <chisoon.jeong@lge.com>
Date:   Thu Apr 28 21:00:50 2022 +0000

    GCC: Fixed build error by adding constexpr to base::ByteSwap(unit32_t)
    
    Without this change build fails on compile step with error:
    
    In file included from //components/variations/hashing.cc:10:
    //base/sys_byteorder.h: In function 'constexpr uintptr_t base::ByteSwapUintPtrT(uintptr_t)':
    //base/sys_byteorder.h:66:20: error: call to non-'constexpr' function 'uint32_t base::ByteSwap(uint32_t)'
       66 |     return ByteSwap(static_cast<uint32_t>(x));
          |            ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
    
    Bug:819294
    
    Change-Id: I05134e02020e38433fe8bbe94f8f6870dac76390
    Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3600155
    Reviewed-by: Nico Weber <thakis@chromium.org>
    Commit-Queue: Nico Weber <thakis@chromium.org>
    Cr-Commit-Position: refs/heads/main@{#997404}

diff --git a/base/sys_byteorder.h b/base/sys_byteorder.h
index b001cc60c5e65..5516be76b13e3 100644
--- a/base/sys_byteorder.h
+++ b/base/sys_byteorder.h
@@ -30,7 +30,7 @@ inline uint16_t ByteSwap(uint16_t x) {
 #endif
 }
 
-inline uint32_t ByteSwap(uint32_t x) {
+inline constexpr uint32_t ByteSwap(uint32_t x) {
 #if defined(COMPILER_MSVC) && !defined(__clang__)
   return _byteswap_ulong(x);
 #else
